<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1130</width>
    <height>727</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>110</y>
      <width>1101</width>
      <height>571</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="processFontTab">
     <attribute name="title">
      <string>Process font</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="displayFontTab">
     <attribute name="title">
      <string>Display font</string>
     </attribute>
     <widget class="QLabel" name="glyphShowLabel">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>371</width>
        <height>381</height>
       </rect>
      </property>
      <property name="text">
       <string>PLACEHOLDER</string>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QComboBox" name="selectGlyphComboBox2">
      <property name="geometry">
       <rect>
        <x>71</x>
        <y>30</y>
        <width>271</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="selectGlyphArrowLeft2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>41</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;</string>
      </property>
     </widget>
     <widget class="QPushButton" name="selectGlyphArrowRight2">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>30</y>
        <width>41</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="shaderTestTab">
     <attribute name="title">
      <string>Shader test</string>
     </attribute>
     <widget class="QPlainTextEdit" name="hlslEditor">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>621</width>
        <height>161</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="glslEditor">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>621</width>
        <height>171</height>
       </rect>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Noto Sans'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;#version 330 core&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;out vec4 FragColor;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;in vec2 TexCoords;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;uniform sampler2D sdf_tex;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;void main()&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;{&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    float sdfValue = texture(sdf_tex, TexCoords).r;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    FragColor = vec4(TexCoords.x, TexCoords.y, 1.0, smoothstep(0.49,0.51,sdfValue) ); &lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;}&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QPushButton" name="glslCompileBtn">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>410</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Compile</string>
      </property>
     </widget>
     <widget class="QPushButton" name="hlsl2GlslBtn">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>200</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Transpile</string>
      </property>
     </widget>
     <widget class="QComboBox" name="selectGlyphComboBox1">
      <property name="geometry">
       <rect>
        <x>701</x>
        <y>10</y>
        <width>331</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="selectGlyphArrowLeft1">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>10</y>
        <width>31</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>&lt;</string>
      </property>
     </widget>
     <widget class="QPushButton" name="selectGlyphArrowRight1">
      <property name="geometry">
       <rect>
        <x>1040</x>
        <y>10</y>
        <width>31</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>&gt;</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="enableHlslBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>82</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>HLSL</string>
      </property>
     </widget>
     <widget class="OpenGLCanvas" name="openGLWidget">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>50</y>
        <width>411</width>
        <height>351</height>
       </rect>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>420</y>
        <width>401</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider_2">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>440</y>
        <width>401</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider_3">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>460</y>
        <width>401</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSlider" name="horizontalSlider_4">
      <property name="geometry">
       <rect>
        <x>660</x>
        <y>480</y>
        <width>401</width>
        <height>16</height>
       </rect>
      </property>
      <property name="value">
       <number>99</number>
      </property>
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="SaveLoadGrp">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1111</width>
      <height>100</height>
     </rect>
    </property>
    <property name="title">
     <string>Save / Load</string>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="loadPathBtn">
       <property name="text">
        <string>Browse Load Path</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="loadPathEdit"/>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="loadBtn">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Load</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="saveBtnPath">
       <property name="text">
        <string>Browse Save path</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="savePathEdit"/>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="saveBtn">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1130</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OpenGLCanvas</class>
   <extends>QOpenGLWidget</extends>
   <header>OpenGLCanvas.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
